server_src/services/database/server-settings.service.js

```diff
@@ -0,0 +1,294 @@
+const ServerSettingsDB = require("../../models/ServerSettings.js");
+
+const { isEmpty } = require("lodash");
+
+class ServerSettings {
+  currentServerSettings;
+  /**
+   * Initialises server settings on boot, reboot and makes sure all required values are defined. Creates cache of current values.
+   * @returns {String} String stating success or failure
+   */
+  async init() {
+    let currentSettings;
+    try {
+      currentSettings = await this.returnDatabaseValues();
+      if (currentSettings.length < 1) {
+        let defaultServerSettings = new ServerSettingsDB({});
+        await defaultServerSettings.save();
+        this.currentServerSettings = defaultServerSettings;
+        return "Default server settings have been created";
+      } else {
+        // Re-save settings to make sure defaults are available...
+        await currentSettings[0].save();
+        this.currentServerSettings = currentSettings[0];
+        return "Server settings already exist, loaded existing values...";
+      }
+    } catch (e) {
+      if (e.message.includes("command find requires authentication")) {
+        throw "Database authentication failed.";
+      } else {
+        throw "Database connection failed.";
+      }
+    }
+  }
+  /**
+   * Resets server settings to default and updates the internal cache
+   */
+  async resetServerSettingsToDefault() {
+    const currentDatabaseSettings = await this.returnDatabaseValues({});
+    await currentDatabaseSettings.deleteMany({});
+    await currentDatabaseSettings.save();
+    this.currentServerSettings = currentDatabaseSettings;
+  }
+  /**
+   * Checks the database for the server settings, can be supplied with filter and options to specify mongoose options/filter.
+   * @param {Object} filter object to filter the database results
+   * @param {Object} options object to change the database options
+   * @returns {Object} Object containing the filtered settings / full settings
+   */
+  returnDatabaseValues({ filter = {}, options = {} } = {}) {
```

Comment:
Function naming unclear to me, suggestion `listServerSettingsDocs`. Oh and btw, I think we're aiming to have 1  server settings, so maybe change it to `getSettings`. Usage would then be `this.getSettings` which is clear to the internal methods here as well to the outside which use the ServerSettings service/repository.

Reply:
A singleton isn't a value though? To me I'm getting the database array of objects.